alacritty_terminal/src/ansi.rs

```diff
@@ -25,62 +25,49 @@ use vte;
 use crate::term::color::Rgb;
 
 // Parse color arguments
-//
-// Expect that color argument looks like "rgb:xx/xx/xx" or "#xxxxxx"
-fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
-    let mut iter = color.iter();
 
-    macro_rules! next {
-        () => {
-            iter.next().map(|v| *v as char)
-        };
+fn is_rgb_separator(ch: Option<char>) -> bool {
+    ch == None || ch == Some('/') || ch == Some('\x07')
+}
+
+// Expect that color argument looks like "rgb:x[xxx]/x[xxx]/x[xxx]" or "#xxxxxx"
+fn parse_rgb_color(color: &[u8]) -> Option<Rgb> {
+    let len = color.len();
+    // Legacy colors are in the form "#xxxxxx"
```

Comment:
This statement is not true. Legacy colors can be in the format `#RGB` up to `#RRRRGGGGBBBB`. This should be handled properly, if we're already trying to improve the XParseColor compatability.

Reply:
Full disclosure, my main goal with this PR is really to support 4-digit color queries, which has already been added to the main branch separate from this issue. Improving XParseColor compatibility was just a prerequisite. However I am happy to help and I have added support for more color formats [here](https://github.com/rbong/alacritty/blob/cdbcd8357fc2955c011785ed02bd93b386509561/alacritty_terminal/src/ansi.rs#L53-L72) with tests [here](https://github.com/rbong/alacritty/blob/cdbcd8357fc2955c011785ed02bd93b386509561/alacritty_terminal/src/ansi.rs#L1547-L1550)